pennylane/return_types.py

```diff
@@ -588,6 +492,147 @@ def circuit_stack(a, b):
+
+        where before the following error was raised:
+
+        .. code-block:: python
+
+            ValueError: All input arrays must have the same shape.
+
+        The new return types system also unlocks the use of shot vectors with all the previous features. For example you
+        can take the second derivative and multiple measurement with with JAX:
+
+        .. code-block:: python
+
+            import jax
+
+            qml.enable_return()
+
+            dev = qml.device("default.qubit", wires=2, shots=(1, 10000))
+
+            params = jax.numpy.array([0.1, 0.2])
+
+            @qml.qnode(dev, interface="jax", diff_method="parameter-shift", max_diff=2)
+            def circuit(x):
+                qml.RX(x[0], wires=[0])
+                qml.RY(x[1], wires=[1])
+                qml.CNOT(wires=[0, 1])
+                return qml.var(qml.PauliZ(0) @ qml.PauliX(1)), qml.probs(wires=[0])
+
+        >>> jax.hessian(circuit)(params)
+        ((Array([[ 0.,  0.],
+                        [ 2., -3.]], dtype=float32),
+          Array([[[-0.5,  0. ],
+                       [ 0. ,  0. ]],
+                      [[ 0.5,  0. ],
+                       [ 0. ,  0. ]]], dtype=float32)),
+         (Array([[ 0.07677898,  0.0563341 ],
+                       [ 0.07238522, -1.830669  ]], dtype=float32),
+          Array([[[-4.9707499e-01,  2.9999996e-04],
+                        [-6.2500127e-04,  1.2500001e-04]],
+                       [[ 4.9707499e-01, -2.9999996e-04],
+                        [ 6.2500127e-04, -1.2500001e-04]]], dtype=float32)))
+
+    Note that failing to set the ``max-diff`` with jitting will raise a somewhat unrelated error:
+
+        .. code-block:: python
+
+            import jax
+            import pennylane as qml
+
+            jax.config.update("jax_enable_x64", True)
+            qml.enable_return()
+
+            dev = qml.device("default.qubit", wires=2, shots=10000)
+
+            params = jax.numpy.array([0.1, 0.2])
+
+            @jax.jit
+            @qml.qnode(dev, interface="jax", diff_method="parameter-shift") # Note: max_diff=2 should be passed here
+            def circuit(x):
+                qml.RX(x[0], wires=[0])
+                qml.RY(x[1], wires=[1])
+                qml.CNOT(wires=[0, 1])
+                return qml.var(qml.PauliZ(0) @ qml.PauliX(1)), qml.probs(wires=[0])
+
+        >>> jax.hessian(circuit)(params)
+        ~/anaconda3/lib/python3.8/site-packages/jax/_src/callback.py in pure_callback_jvp_rule(***failed resolving arguments***)
+             53 def pure_callback_jvp_rule(*args, **kwargs):
+             54   del args, kwargs
+        ---> 55   raise ValueError(
+             56       "Pure callbacks do not support JVP. "
+             57       "Please use `jax.custom_jvp` to use callbacks while taking gradients.")
+
+        ValueError: Pure callbacks do not support JVP. Please use `jax.custom_jvp` to use callbacks while taking gradients.
```

Comment:
It seems Sphinx doesn't like newlines in tracebacks, and renders this as outside of the console output. I think we just have to accept this limitation and remove the newline here.